apps/web/src/components/layout/components/HeaderNavNew.tsx

```diff
@@ -0,0 +1,230 @@
+    icon: <IconSettings color={iconColor} />,
+    path: ROUTES.SETTINGS,
+  },
+  {
+    title: 'Invite Members',
+    icon: <IconOutlineGroupAdd color={iconColor} />,
+    path: ROUTES.TEAM,
+  },
+];
+const headerIconsSettings = { color: colors.B60, size: 20 };
+
+const Icon = () => {
+  const { themeStatus } = useLocalThemePreference();
+
+  if (themeStatus === 'dark') {
+    return <IconDarkMode {...headerIconsSettings} />;
+  }
+  if (themeStatus === 'light') {
+    return <IconLightMode {...headerIconsSettings} />;
+  }
+
+  return <IconTonality {...headerIconsSettings} />;
+};
+
+export function HeaderNavNew() {
+  const { currentOrganization, currentUser, logout } = useAuthContext();
+  const { colorScheme, toggleColorScheme } = useMantineColorScheme();
+  const { themeStatus } = useLocalThemePreference();
+  const { addItem, removeItems } = useSpotlightContext();
+  const { boot } = useIntercom();
+  const segment = useSegment();
+  const isSelfHosted = IS_DOCKER_HOSTED;
+  const isDark = colorScheme === 'dark';
+  const iconColor = isDark ? colors.white : colors.B40;
+
+  const debounceThemeChange = useDebounce((args: { colorScheme: ColorScheme; themeStatus: string }) => {
+    segment.track('Theme is set - [Theme]', args);
+  }, 500);
+
+  useEffect(() => {
+    debounceThemeChange({ colorScheme, themeStatus });
+  }, [colorScheme, themeStatus, debounceThemeChange]);
+
+  useEffect(() => {
+    const shouldBootIntercom = !!INTERCOM_APP_ID && currentUser && currentOrganization;
+    if (shouldBootIntercom) {
+      boot({
+        userId: currentUser._id,
+        email: currentUser?.email ?? '',
+        name: currentUser?.firstName + ' ' + currentUser?.lastName,
+        createdAt: currentUser?.createdAt,
+        company: {
+          name: currentOrganization?.name,
+          companyId: currentOrganization?._id as string,
+        },
+        userHash: currentUser.servicesHashes?.intercom,
+        customLauncherSelector: '#intercom-launcher',
+        hideDefaultLauncher: true,
+      });
+    }
+  }, [boot, currentUser, currentOrganization]);
+
+  let themeTitle = 'Match System Appearance';
+  if (themeStatus === 'dark') {
+    themeTitle = 'Dark Theme';
+  } else if (themeStatus === 'light') {
+    themeTitle = 'Light Theme';
+  }
+
+  const additionalMenuItems = useMemo(() => {
+    return [
+      {
+        id: 'toggle-theme',
+        title: themeTitle,
+        icon: <Icon />,
+        onTrigger: () => {
+          toggleColorScheme();
+        },
+      },
+      {
+        id: 'sign-out',
+        title: 'Sign out',
+        icon: <IconLogout />,
+        onTrigger: () => {
+          logout();
+        },
+      },
+    ];
+  }, [toggleColorScheme, logout, themeTitle]);
+
+  useEffect(() => {
+    removeItems(additionalMenuItems.map((item) => item.id));
+
+    addItem(additionalMenuItems);
+  }, [addItem, removeItems, additionalMenuItems]);
+
+  const menuItems = useMemo(() => menuItem(iconColor), [iconColor]);
```

Comment:
These are fed into the Spotlight provider.
I think without the useMemo/useEffect it will re-render the whole app multiple times as it is a part of the context provider.